docs/guides/initializers.md

```diff
@@ -0,0 +1,123 @@
+
++++ {"id": "sG5oygwB7UrW"}
+
+Note that even if the `Initializer` is deterministic, **a PRNGKey must be passed**. Therefore functions like [`jax.numpy.zeros`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.zeros.html) and [`jax.numpy.ones`](https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.ones.html) can't be used as initializers as they have an invalid function signature.
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 60
+  status: ok
+  timestamp: 1673133041047
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: IjLW1abzw2pM
+outputId: 2616d907-7e24-4cd3-e926-3c39c8dd5401
+---
+layer = nn.Dense(features=5, kernel_init=jnp.zeros) # jnp.zeros has an invalid function signature
+try:
+  layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params'] # <-- throws an error
+except Exception as e:
+  print(f'Caught error: {e}')
+```
+
+```{code-cell}
+---
+executionInfo:
+  elapsed: 62
+  status: ok
+  timestamp: 1673133041242
+  user:
+    displayName: Marcus Chiam
+    userId: '17531616275590396120'
+  user_tz: 480
+id: RlOIOCj68p86
+outputId: 5871f462-1d24-47f9-cc20-6fa7f358ea99
+---
+layer = nn.Dense(features=5, kernel_init=zeros) # use flax.linen.initializers.zeros instead
+layer.init(jax.random.PRNGKey(42), jnp.empty((1,4)))['params'] # <-- successfully generate params
+```
+
++++ {"id": "S4X_xHHk-b4V"}
+
+## `Initializer` restrictions for `bias_init`
+
++++ {"id": "jLUFaliMy_DT"}
+
+Only the `Initializer` functions built by [`constant`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.constant.html), [`normal`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.normal.html) and [`uniform`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.uniform.html) and the `Initializer` functions [`zeros`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.zeros.html) and [`ones`](https://flax.readthedocs.io/en/latest/api_reference/_autosummary/flax.linen.initializers.ones.html) can be used as valid arguments for `bias_init`.
```

Comment:
In principle we could use any function we want, but we just cannot use the ones you listed right? So I would remove the word "Only" here.